scm-webapp/src/main/java/sonia/scm/api/v2/resources/RepositoryImportResource.java

```diff
@@ -0,0 +1,548 @@
+//    } catch (WebApplicationException ex) {
+//      logger.warn("error durring bundle import", ex);
+//      response = Response.fromResponse(ex.getResponse()).entity(
+//        new RestActionUploadResult(false)).build();
+//    }
+//
+//    return response;
+//  }
+
+  /**
+   * Imports a external repository which is accessible via url. The method can
+   * only be used, if the repository type supports the {@link Command#PULL}. The
+   * method will return a location header with the url to the imported
+   * repository.
+   *
+   * @param uriInfo uri info
+   * @param type    repository type
+   * @param request request object
+   * @return empty response with location header which points to the imported
+   * repository
+   * @since 2.11.0
+   */
+  @POST
+  @Path("{type}/url")
+  @Consumes(VndMediaType.REPOSITORY)
+  @Operation(summary = "Import repository from url", description = "Imports the repository for the given url.", tags = "Repository")
+  @ApiResponse(
+    responseCode = "201",
+    description = "Repository import was successful",
+    content = @Content(
+      mediaType = VndMediaType.REPOSITORY,
+      schema = @Schema(implementation = RepositoryDto.class)
+    )
+  )
+  @ApiResponse(
+    responseCode = "401",
+    description = "not authenticated / invalid credentials"
+  )
+  @ApiResponse(
+    responseCode = "403",
+    description = "not authorized, the current user has no privileges to read the repository"
+  )
+  @ApiResponse(
+    responseCode = "500",
+    description = "internal server error",
+    content = @Content(
+      mediaType = VndMediaType.ERROR_TYPE,
+      schema = @Schema(implementation = ErrorDto.class)
+    )
+  )
+  public Response importFromUrl(@Context UriInfo uriInfo,
+                                @PathParam("type") String type, RepositoryImportDto request) {
+    RepositoryPermissions.create().check();
+    checkNotNull(request, "request is required");
+    checkArgument(!Strings.isNullOrEmpty(request.getName()),
+      "request does not contain name of the repository");
+    checkArgument(!Strings.isNullOrEmpty(request.getImportUrl()),
+      "request does not contain url of the remote repository");
+
+    Type t = type(type);
+
+    checkSupport(t, Command.PULL, request);
+
+    logger.info("start {} import for external url {}", type, request.getImportUrl());
+
+    Repository repository = new Repository(null, type, request.getNamespace(), request.getName());
+    try {
+      repository = manager.create(
+        repository,
+        pullChangesFromRemoteUrl(request)
+      );
+    } catch (Exception e) {
+      eventBus.post(new RepositoryImportEvent(HandlerEventType.MODIFY, repository, true));
+      throw e;
+    }
+    eventBus.post(new RepositoryImportEvent(HandlerEventType.MODIFY, repository, false));
+
+    return Response.created(URI.create(resourceLinks.repository().self(repository.getNamespace(), repository.getName()))).build();
```

Comment:
Moving this lines into the try block, would make it more clear that this is the success case.